lib/core/src/irods_service_account.cpp

```diff
@@ -0,0 +1,132 @@
+#include "irods/irods_service_account.hpp"
+
+#include "irods/irods_exception.hpp"
+#include "irods/irods_get_full_path_for_config_file.hpp"
+#include "irods/irods_kvp_string_parser.hpp"
+#include "irods/irods_log.hpp"
+#include "irods/rodsErrorTable.h"
+
+#include <cerrno>
+#include <cstdint>
+#include <cstdlib>
+#include <fstream>
+#include <memory>
+
+#include <pwd.h>
+#include <sys/types.h>
+#include <unistd.h>
+
+#include <boost/filesystem.hpp>
+#include <boost/algorithm/string.hpp>
+#include <fmt/format.h>
+
+namespace irods
+{
+    const std::string SERVICE_ACCOUNT_CONFIG_FILE("service_account.config");
+    const std::string SERTICE_ACCOUNT_KW("IRODS_SERVICE_ACCOUNT_NAME");
+
+    const size_t PWD_BUF_LEN_DEFAULT = 16384; // default size of buffer for strings in struct passwd
+    const int PWD_BUF_LEN_MINIMUM = 128; // minimum size of buffer for strings in struct passwd
+
+    // FIXME: Parsing of service_account.config isn't super robust.
+    auto is_service_account() -> bool
+    {
+        std::string service_account_username;
+
+        std::string svc_acct_cfg_fn;
+        irods::error ret = irods::get_full_path_for_config_file(SERVICE_ACCOUNT_CONFIG_FILE, svc_acct_cfg_fn);
+
+        if (!ret.ok()) {
+            // if config file not found, the machine probably does not have a server
+            return false;
+        }
+
+        std::ifstream svc_acct_cfg_strm{svc_acct_cfg_fn};
+        if (!svc_acct_cfg_strm.is_open()) {
+            // if we can't open the config file, it's likely due to a permission issue
+            // meaning it is extremely unlikely we are the service account
+            return false;
+        }
+
+        kvp_map_t service_acct_cfg;
+        uint_fast64_t cfg_line_no = 0;
```

Comment:
Why use `uint_fast64_t` instead of `std::uint64_t`?

Reply:
It just seemed like the thing to do. I can change it

Reply:
Yeah, let's go with a more common type. If switching to std::uint64_t costs us something, we can address it later.